intrahospital_api/management/commands/batch_load2.py

```diff
@@ -63,13 +66,15 @@ def handle(self, *args, **options):
                 hospital_number=item['demographics']["hospital_number"]
             )
 
+            if not item['demographics']["hospital_number"]:
+                continue
+
             if not patient_demographics_set.exists():
                 continue # Not in our cohort
 
-            update_lab_tests.update_tests(
-                patient_demographics_set.first().patient,
-                item["lab_tests"],
-            )
+            item['lab_tests'] = [i for i in item['lab_tests'] if i["test_name"] or i["external_identifier"]]
```

Comment:
Can we make this _much_ clearer e.g. definitely not in a list comprehension, and with comments